lib/ardb.rb

```diff
@@ -57,30 +50,34 @@ def self.require_db_file
   end
 
   class Config
-    include NsOptions::Proxy
-
-    namespace :db do
-      option :adapter,          String,  :required => true
-      option :database,         String,  :required => true
-      option :encoding,         String,  :required => false
-      option :host,             String,  :required => false
-      option :port,             Integer, :required => false
-      option :username,         String,  :required => false
-      option :password,         String,  :required => false
-      option :pool,             Integer, :required => false
-      option :checkout_timeout, Integer, :required => false
-    end
-
-    option :logger,                :required => true
-    option :schema_format, Symbol, :default => :ruby
 
-    attr_accessor :root_path
+    ACTIVERECORD_ATTRS = [
+      :adapter,
+      :database,
+      :encoding,
+      :host,
+      :port,
+      :username,
+      :password,
+      :pool,
+      :checkout_timeout
+    ].freeze
+    DEFAULT_MIGRATIONS_PATH = 'db/migrations'.freeze
+    DEFAULT_SCHEMA_PATH     = 'db/schema'.freeze
+    DEFAULT_SCHEMA_FORMAT   = :ruby
+    VALID_SCHEMA_FORMATS    = [DEFAULT_SCHEMA_FORMAT, :sql].freeze
+
+    attr_accessor *ACTIVERECORD_ATTRS
+    attr_accessor :logger, :root_path
+    attr_reader :schema_format
     attr_writer :migrations_path, :schema_path
 
     def initialize
+      @logger          = Logger.new(STDOUT)
```

Comment:
@kellyredding - I chose to default it and make it optional (you can set it to `nil` to disable logging) instead of defaulting it to `nil` and making it required. I think previously if you didn't set it it would error. I felt it was a better default behavior to have logging to stdout by default and not force you to set a logger. I could also be persuaded to default it to `nil` and still leave it optional. This would mean logging would be disabled by default.